lib/parser/source/buffer.rb

```diff
@@ -228,13 +228,15 @@ def line_begins
         @line_begins
       end
 
-      def line_for(position)
-        if line_begins.respond_to? :bsearch
+      if [].respond_to?(:bsearch)
```

Comment:
Maybe `Array.instance_methods.include?(:bsearch)` instead? A bit more nicer to read than the indirection through an instance. Performance wise not relevant.